packages/expo-auth-session/src/AuthRequestHooks.ts

```diff
@@ -40,23 +45,38 @@ export function useAuthRequest(
 ] {
   const [request, setRequest] = useState<AuthRequest | null>(null);
   const [result, setResult] = useState<AuthSessionResult | null>(null);
+  const isMountedRef = useRef<boolean | null>(null)
+  
+  useEffect(() => {
+    isMountedRef.current = true
+    
+    return () => {
+      isMountedRef.current = false
+    }
+  }, [])
 
   const promptAsync = useCallback(
     async (options: AuthRequestPromptOptions = {}) => {
       if (!discovery || !request) {
         throw new Error('Cannot prompt to authenticate until the request has finished loading.');
       }
       const result = await request?.promptAsync(discovery, options);
-      setResult(result);
+      isMountedRef.current && setResult(result);
       return result;
     },
     [request?.url, discovery?.authorizationEndpoint]
   );
 
   useEffect(() => {
+    let stillCareAboutThis = true
+      
     if (config && discovery) {
       const request = new AuthRequest(config);
-      request.makeAuthUrlAsync(discovery).then(() => setRequest(request));
+      request.makeAuthUrlAsync(discovery).then(() => stillCareAboutThis && setRequest(request));
+    }
+      
+    return () => { 
+      stillCareAboutThis = false 
```

Comment:
Make sure to run the linter over everything before submitting a PR for review.

Reply:
Please realise that the 2GB repository is pretty privileged to _have_ to download in order to make a preliminary PR -- which as mentioned I made from my phone.

Would it be okay if the next time I just mark it as a draft so that I can run the linter when I'm at a desktop and not 'waste your time' persé?

Reply:
What ultimately matters is that Expo is good for the people using it. You can submit a draft PR but you probably do need to test your code as one of the ways to help make sure it's good. Also, CI will run the linter amongst other checks but it's usually nicer to get that feedback immediately on your own computer.